addons/onboarding/models/onboarding_progress.py

```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+from odoo.addons.onboarding.models.onboarding_progress_step import ONBOARDING_PROGRESS_STEP_STATES
+
+
+ONBOARDING_PROGRESS_STATES = ONBOARDING_PROGRESS_STEP_STATES + [('closed', "Closed")]
+
+
+class OnboardingProgress(models.Model):
+    _name = 'onboarding.progress'
+    _description = 'Onboarding Progress Tracker'
+
+    state = fields.Selection(
+        ONBOARDING_PROGRESS_STATES, string='Onboarding progress', compute='_compute_state', store=True)
+    closed = fields.Boolean('Was panel closed?')
+    company_id = fields.Many2one('res.company')
+    onboarding_id = fields.Many2one('onboarding.onboarding', 'Related onboarding tracked', ondelete='cascade')
+    progress_step_ids = fields.One2many('onboarding.progress.step', 'progress_id', 'Progress Steps Trackers')
+    _sql_constraints = [
+        ('onboarding_company_uniq', 'unique (onboarding_id,company_id)',
+         'There cannot be multiple records of the same onboarding completion for the same company.'),
+    ]
+
+    @api.depends('progress_step_ids.state')
+    def _compute_state(self):
+        progress_steps_data = self.env['onboarding.progress.step'].read_group(
+            [('progress_id', 'in', self.ids), ('state', '=', 'not_done')],
+            ['progress_id'], ['progress_id']
+        )
+        result = dict((data['progress_id'][0], data['progress_id_count']) for data in progress_steps_data)
+        for progress in self:
+            progress.state = 'not_done' if result.get(progress.id) else 'done'
+
+    def get_and_update_onboarding_states(self):
+        self.ensure_one()
+        old_values = {}
+        all_done = True
+        for progress_step in self.progress_step_ids:
+            old_values[progress_step.step_id.name] = progress_step.state
+            progress_step.consolidate_just_done()
+            if progress_step.state not in {'just_done', 'done'}:
```

Comment:
Can the state even be "just_done" at this point since we just consolidated?

Also, can't we just check the "current step", instead of consolidating all of them?